build/Dockerfile

```diff
@@ -1,18 +1,10 @@
 # syntax=docker/dockerfile:1.4
 FROM golang:1.21 as builder
-ARG VERSION
-ARG GIT_COMMIT
-ARG DATE
 
-WORKDIR /go/src/github.com/nginxinc/nginx-kubernetes-gateway/cmd/gateway
+WORKDIR /go/src/github.com/nginxinc/nginx-kubernetes-gateway
 
-COPY go.mod go.sum /go/src/github.com/nginxinc/nginx-kubernetes-gateway
-RUN go mod download
```

Comment:
Sorry just noticing this, we should keep these two lines so we can cache the go modules and not download them every time.

Reply:
IIRC it needs to be a separate `COPY` to cache it properly

Reply:
btw I'm not 100% sure of this, I'll investigate 🔬